kroxylicious-filters/kroxylicious-record-encryption/src/test/java/io/kroxylicious/filter/encryption/dek/DekManagerTest.java

```diff
@@ -98,4 +102,18 @@ void testDecryptedEdekIsGoodForDecryptingData(CipherManager cipherManager) {
         assertThat(new String(decodedPlaintext.array(), StandardCharsets.UTF_8)).isEqualTo("hello, world");
     }
 
+    @Test
+    void aes256KeyMustBe256bits() {
+        var fixedDekKmsService = new FixedDekKmsService(128);
+        DekManager<ByteBuffer, ByteBuffer> manager = new DekManager<>(fixedDekKmsService, new FixedDekKmsService.Config(), 10000);
+        CompletionStage<Dek<ByteBuffer>> dekCompletionStage = manager.generateDek(fixedDekKmsService.getKekId(), Aes.AES_256_GCM_128);
+        assertThatThrownBy(() -> {
+            dekCompletionStage.toCompletableFuture().join();
+        }).isInstanceOf(CompletionException.class)
+                .cause()
+                .isExactlyInstanceOf(EncryptionConfigurationException.class)
+                .hasMessageContaining("KMS returned 128-bit DEK but AES_256_GCM_128 requires keys of 256 bits");
```

Comment:
sonar doesn't like calling `toCompelteableFuture` and `join` in the `assertThatThrownBy` so how about 
```suggestion
        assertThat(dekCompletionStage).failsWithin(10, TimeUnit.SECONDS)
                .withThrowableOfType(ExecutionException.class)
                .havingCause()
                .isExactlyInstanceOf(EncryptionConfigurationException.class)
                .withMessage("KMS returned 128-bit DEK but AES_256_GCM_128 requires keys of 256 bits");
```

Reply:
The `withThrowableOfType(ExecutionException.class)` is arguably excessive.